IHP/QueryBuilder.hs

```diff
@@ -1020,6 +1030,26 @@ queryUnion firstQueryBuilderProvider secondQueryBuilderProvider = NoJoinQueryBui
     
 {-# INLINE queryUnion #-}
 
+-- | Like 'queryUnion', but applied on all the elements on the list
+--
+-- >  action ProjectsAction = do
+-- >      let values :: [(ProjectType, Int)] = [(ProjectTypeOngoing, 3), (ProjectTypeNotStarted, 2)]
+-- >
+-- >          valuePairToCondition :: (ProjectType, Int) -> QueryBuilder "projects"
+-- >          valuePairToCondition (projectType, participants) =
+-- >              query @Project
+-- >                  |> filterWhere (#projectType, projectType)
+-- >                  |> filterWhere (#participants, participants)
+-- >
+-- >          theQuery = queryOrList (map valuePairToCondition values)
+-- >
+-- >      projects <- fetch theQuery
+-- >      render IndexView { .. }
+queryUnionList :: forall table. (Table (GetModelByTableName table), KnownSymbol table, GetTableName (GetModelByTableName table) ~ table) => [QueryBuilder table] -> QueryBuilder table
+queryUnionList [] = FilterByQueryBuilder { queryBuilder = query @(GetModelByTableName table) @table, queryFilter = ("id", NotEqOp, Plain "id"), applyLeft = Nothing, applyRight = Nothing }
```

Comment:
With https://github.com/digitallyinduced/ihp/pull/1925, is `"id"` correct?  (That is, are we guarnteed to havd an `record.id` ?)

Reply:
Good point. Likely we should use `head (primaryKeyColumnNames @(GetModelByTableName table))` here. Or even better support free form SQL snippets here (we just want `WHERE false` here, but there's no way to write this with the current query builder)